src/clj/runbld/build.clj

```diff
@@ -37,12 +39,99 @@
        :job-name-extra job-name-extra
        :org-project-branch (format "%s/%s#%s" org project branch)})))
 
+(defn find-build [opts id]
+  (when id
+    (-> (doc/search (-> opts :es :conn)
+                    (-> opts :es :build-index-search)
+                    {:body
+                     {:query
+                      {:match
+                       {:id id}}}})
+        :hits :hits first :_source)))
+
+(defn query-for-build [keyword-mapping? job-name vcs-provider]
+  (let [clauses (if keyword-mapping?
+                  ;; pre-5.0 mapping for string fields
+                  [{:term {:build.job-name job-name}}
+                   {:term {:process.status "SUCCESS"}}
+                   {:term {:vcs.provider vcs-provider}}]
+                  ;; post-5.0 mapping for string fields, before the
+                  ;; mapping was updated in the code
+                  [{:term {:build.job-name.keyword job-name}}
+                   {:term {:process.status.keyword "SUCCESS"}}
+                   {:term {:vcs.provider.keyword vcs-provider}}])]
+    {:body
+     {:query
+      {:bool
+       {:filter clauses}},
+      ;; search for time-start, because we want the latest commit that
+      ;; has a *finished* job
```

Comment:
Good catch. I'd forgotten that we haven't yet implemented https://github.com/elastic/runbld/issues/27.